lib/active_record_doctor/detectors/missing_unique_indexes.rb

```diff
@@ -42,7 +48,27 @@ def detect
             columns = (scope + validator.attributes).map(&:to_s)
             next if ignore_columns.include?("#{model.name}(#{columns.join(',')})")
 
-            problem!(table: model.table_name, columns: columns)
+            problem!(model: model, table: model.table_name, columns: columns, problem: :validations)
+          end
+        end
+      end
+
+      def has_ones_without_indexes # rubocop:disable Naming/PredicateName
+        models.each do |model|
+          has_ones = model.reflect_on_all_associations(:has_one)
+          has_ones.each do |has_one|
+            next if has_one.is_a?(ActiveRecord::Reflection::ThroughReflection) || has_one.scope
+
+            association_model = has_one.klass
+            next if config(:ignore_models).include?(association_model.name)
```

Comment:
I think I understand your intent here. However, using `ignore_models` like that deviates from how it's used in all other detectors and might be confusing to users: adding a model to the ignored list would ignore its validations but **not** it's `has_one` associations.

I suggest we use `ignore_models` like in all other detectors (i.e. `models(except: config(:ignore_models)).each ...`). If the user wants to silence a warning he should use `ignore_columns` - that would be consistent with how it works in the case of validations. When the users sees:

```
add a unique index on users(organization_id) - ...
```

then he knows this can be ignored by adding `"users.organization_id"` to `ignored_columns`.